src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs

```diff
@@ -685,6 +685,12 @@ public void Dispose()
             _upgradeTimer?.Dispose();
         }
 
+        private void UpdatePeerCountMetric(NodeClientType clientType)
+        {
+            long count = _peers.Count(k => k.Value.PeerClientType == clientType);
+            Metrics.SyncPeers[clientType] = count;
```

Comment:
Can we just increment/Decrement in dictionary instead of doing count every time?

Reply:
I thought of that originally, but I'm concerned that some update might get missed causing faulty metric.